langchain4j-core/src/main/java/dev/langchain4j/model/chat/response/StreamingChatResponseHandler.java

```diff
@@ -0,0 +1,47 @@
+package dev.langchain4j.model.chat.response;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.model.chat.StreamingChatLanguageModel;
+
+/**
+ * TODO review all javadoc in this class
+ * Represents a handler for streaming responses from a {@link StreamingChatLanguageModel}.
+ * The handler is invoked each time the model generates a new token in a textual response.
+ * If the model executes a tool instead, {@link #onCompleteResponse} will be invoked instead.
+ *
+ * @see StreamingChatLanguageModel
+ */
+public interface StreamingChatResponseHandler {
```

Comment:
Would be nice to adapt the convenient lambda streaming response handler that I had created to make it work with the new response handler.

Reply:
Good point, thanks!